### PR TITLE
fix qvalue mask_action error for obs_next

### PR DESCRIPTION
**Pull request by Trinkle23897**

### CODE REVIEW

**Review by danagi — COMMENTED** [review]
I think the problem in #309 is caused by the function `_target_q` which uses unmasked logits to get target max q values in dqn.py. Since self(...).act is the legal max q value action, my personal suggestion is as follow:
```
model = 'model_old' if self._target else 'model'
result = self(batch, model=model, input='obs_next')
a = result.act
target_q = result.logits[np.arange(len(a)), a]
```

**Review by danagi — APPROVED** [review]
